File: Program/ActEditor/bin/pl/wroc/pwr/student/acteditor/parsing/xsd/Tokenizer.js
Clazz.declarePackage ("pl.wroc.pwr.student.acteditor.parsing.xsd");
c$ = Clazz.declareType (pl.wroc.pwr.student.acteditor.parsing.xsd, "Tokenizer");
Clazz.defineMethod (c$, "getEvent", 
function (line) {
if (this.containsDocumentation (line)) {
return "documentation";
} else if (this.containsElementDefinition (line)) {
return "element-def";
} else if (this.containsOneLineElementDefinition (line)) {
return "one-line-element-def";
} else if (this.containsClosingElementDefinition (line)) {
return "close-element-def";
} else if (this.containsElementReferenceWithoutClosingSign (line)) {
return "element-ref-without-closing";
} else if (this.containsOneLineElementReference (line)) {
return "one-line-element-ref";
} else if (this.containsSequence (line)) {
return "sequence";
} else if (this.containsClosingSequence (line)) {
return "closing-sequence";
} else if (this.containsChoice (line)) {
return "choice";
} else if (this.containsClosingChoice (line)) {
return "closing-choice";
} else if (this.containsGroupDefinition (line)) {
return "element-group-def";
} else if (this.containstGroupReference (line)) {
return "element-group-ref";
} else if (this.containsClosingGroupDefinition (line)) {
return "closing-element-group-def";
} else if (this.containstComplexTypeDefinition (line)) {
return "complex-type-def";
} else if (this.containsClosingComplexTypeDefinition (line)) {
return "closing-complex-type-def";
} else if (this.containsComplexContent (line)) {
return "complex-content";
} else if (this.containsClosingComplexContent (line)) {
return "closing-complex-content";
} else if (this.containsAttributeDefinitionWithoutClosingSign (line)) {
return "attribute-def-without-closing";
} else if (this.containsOneLineAttributeDefinition (line)) {
return "one-line-attribute-def";
} else if (this.containsClosingAttributeDefinition (line)) {
return "closing-attribute-def";
} else if (this.containsAttributeGroupDefinition (line)) {
return "attribute-group-def";
} else if (this.containsAttributeGroupReference (line)) {
return "attribute-group-ref";
} else if (this.containsClosingAttributeGroup (line)) {
return "closing-attribute-group";
} else if (this.containsSimpleTypeDefinition (line)) {
return "simple-type-def";
} else if (this.containsSimpleTypeBeginning (line)) {
return "simple-type-beg";
} else if (this.containsClosingSimpleType (line)) {
return "closing-simple-type";
} else if (this.containsExtension (line)) {
return "extension";
} else if (this.containsClosingExtension (line)) {
return "closing-extension";
} else if (this.containsPattern (line)) {
return "pattern";
} else if (this.containsEnumeration (line)) {
return "enumeration";
}return "ignore";
}, "~S");
Clazz.defineMethod (c$, "containsEnumeration", 
($fz = function (line) {
return line.contains ("<xsd:enumeration");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsPattern", 
($fz = function (line) {
return line.contains ("<xsd:pattern");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsClosingExtension", 
($fz = function (line) {
return line.contains ("</xsd:extension");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsExtension", 
($fz = function (line) {
return line.contains ("<xsd:extension");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsClosingSimpleType", 
($fz = function (line) {
return line.contains ("</xsd:simpleType>");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsSimpleTypeBeginning", 
($fz = function (line) {
return line.contains ("<xsd:simpleType>");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsSimpleTypeDefinition", 
($fz = function (line) {
return (line.contains ("<xsd:simpleType") && line.contains ("name="));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsClosingAttributeGroup", 
($fz = function (line) {
return (line.contains ("</xsd:attributeGroup>"));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsAttributeGroupReference", 
($fz = function (line) {
return (line.contains ("attributeGroup") && line.contains ("ref="));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsAttributeGroupDefinition", 
($fz = function (line) {
return (line.contains ("attributeGroup") && line.contains ("name="));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsOneLineAttributeDefinition", 
($fz = function (line) {
return (line.contains ("attribute ") && line.contains ("name=") && line.contains ("/>"));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsClosingAttributeDefinition", 
($fz = function (line) {
return line.contains ("</xsd:attribute>");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsAttributeDefinitionWithoutClosingSign", 
($fz = function (line) {
return (line.contains ("attribute ") && line.contains ("name=") && !line.contains ("/>"));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsClosingComplexContent", 
($fz = function (line) {
return line.contains ("</xsd:complexContent>");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsComplexContent", 
($fz = function (line) {
return line.contains ("complexContent");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsClosingComplexTypeDefinition", 
($fz = function (line) {
return (line.contains ("</xsd:complexType>"));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containstComplexTypeDefinition", 
($fz = function (line) {
return (line.contains ("complexType") && line.contains ("name="));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containstGroupReference", 
($fz = function (line) {
return (line.contains ("group") && !line.contains ("attributeGroup") && line.contains ("ref="));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsClosingGroupDefinition", 
($fz = function (line) {
return line.contains ("</xsd:group>");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsGroupDefinition", 
($fz = function (line) {
return (line.contains ("group") && !line.contains ("attributeGroup") && line.contains ("name="));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsClosingChoice", 
($fz = function (line) {
return line.contains ("</xsd:choice>");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsChoice", 
($fz = function (line) {
return line.contains ("<xsd:choice");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsClosingSequence", 
($fz = function (line) {
return line.contains ("</xsd:sequence>");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsSequence", 
($fz = function (line) {
return line.contains ("<xsd:sequence");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsOneLineElementDefinition", 
($fz = function (line) {
return (line.contains ("element") && line.contains ("name=") && line.contains ("/>"));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsOneLineElementReference", 
($fz = function (line) {
return (line.contains ("element") && line.contains ("ref=") && line.contains ("/>"));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsElementReferenceWithoutClosingSign", 
($fz = function (line) {
return (line.contains ("element") && line.contains ("ref=") && !line.contains ("/>"));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsClosingElementDefinition", 
($fz = function (line) {
return line.contains ("</xsd:element>");
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsElementDefinition", 
($fz = function (line) {
return (line.contains ("element") && line.contains ("name=") && !line.contains ("/>"));
}, $fz.isPrivate = true, $fz), "~S");
Clazz.defineMethod (c$, "containsDocumentation", 
($fz = function (line) {
return line.contains ("documentation");
}, $fz.isPrivate = true, $fz), "~S");

File: Program/ActEditor/src/pl/wroc/pwr/student/acteditor/parsing/xsd/Tokenizer.java
package pl.wroc.pwr.student.acteditor.parsing.xsd;

public class Tokenizer {
	public String getEvent(String line) {
		if (containsDocumentation(line)) {
			return "documentation";
		} else if (containsElementDefinition(line)) {
			return "element-def";
		} else if (containsOneLineElementDefinition(line)) {
			return "one-line-element-def";
		} else if (containsClosingElementDefinition(line)) {
			return "close-element-def";
		} else if (containsElementReferenceWithoutClosingSign(line)) {
			return "element-ref-without-closing";
		} else if (containsOneLineElementReference(line)) {
			return "one-line-element-ref";
		} else if (containsSequence(line)) {
			return "sequence";
		} else if (containsClosingSequence(line)) {
			return "closing-sequence";
		} else if (containsChoice(line)) {
			return "choice";
		} else if (containsClosingChoice(line)) {
			return "closing-choice";
		} else if (containsGroupDefinition(line)) {
			return "element-group-def";
		} else if (containstGroupReference(line)) {
			return "element-group-ref";
		} else if (containsClosingGroupDefinition(line)) {
			return "closing-element-group-def";
		} else if (containstComplexTypeDefinition(line)) {
			return "complex-type-def";
		} else if (containsClosingComplexTypeDefinition(line)) {
			return "closing-complex-type-def";
		} else if (containsComplexContent(line)) {
			return "complex-content";
		} else if (containsClosingComplexContent(line)) {
			return "closing-complex-content";
		} else if (containsAttributeDefinitionWithoutClosingSign(line)) {
			return "attribute-def-without-closing";
		} else if (containsOneLineAttributeDefinition(line)) {
			return "one-line-attribute-def";
		} else if (containsClosingAttributeDefinition(line)) {
			return "closing-attribute-def";
		} else if (containsAttributeGroupDefinition(line)) {
			return "attribute-group-def";
		} else if (containsAttributeGroupReference(line)) {
			return "attribute-group-ref";
		} else if (containsClosingAttributeGroup(line)) {
			return "closing-attribute-group";
		} else if (containsSimpleTypeDefinition(line)) {
			return "simple-type-def";
		} else if (containsSimpleTypeBeginning(line)) {
			return "simple-type-beg";
		} else if (containsClosingSimpleType(line)) {
			return "closing-simple-type";
		} else if (containsExtension(line)) {
			return "extension";
		} else if (containsClosingExtension(line)) {
			return "closing-extension";
		} else if (containsPattern(line)) {
			return "pattern";
		} else if (containsEnumeration(line)) {
			return "enumeration";
		}
		return "ignore";
	}

	private boolean containsEnumeration(String line) {
		return line.contains("<xsd:enumeration");
	}

	private boolean containsPattern(String line) {
		return line.contains("<xsd:pattern");
	}

	private boolean containsClosingExtension(String line) {
		return line.contains("</xsd:extension");
	}

	private boolean containsExtension(String line) {
		return line.contains("<xsd:extension");
	}

	private boolean containsClosingSimpleType(String line) {
		return line.contains("</xsd:simpleType>");
	}

	private boolean containsSimpleTypeBeginning(String line) {
		return line.contains("<xsd:simpleType>");
	}

	private boolean containsSimpleTypeDefinition(String line) {
		return (line.contains("<xsd:simpleType") && line.contains("name="));
	}

	private boolean containsClosingAttributeGroup(String line) {
		return (line.contains("</xsd:attributeGroup>"));
	}

	private boolean containsAttributeGroupReference(String line) {
		return (line.contains("attributeGroup") && line.contains("ref="));
	}

	private boolean containsAttributeGroupDefinition(String line) {
		return (line.contains("attributeGroup") && line.contains("name="));
	}

	private boolean containsOneLineAttributeDefinition(String line) {
		return (line.contains("attribute ") && line.contains("name=") && line.contains("/>"));
	}

	private boolean containsClosingAttributeDefinition(String line) {
		return line.contains("</xsd:attribute>");
	}

	private boolean containsAttributeDefinitionWithoutClosingSign(String line) {
		return (line.contains("attribute ") && line.contains("name=") && !line.contains("/>"));
	}

	private boolean containsClosingComplexContent(String line) {
		return line.contains("</xsd:complexContent>");
	}

	private boolean containsComplexContent(String line) {
		return line.contains("complexContent");
	}

	private boolean containsClosingComplexTypeDefinition(String line) {
		return (line.contains("</xsd:complexType>"));
	}

	private boolean containstComplexTypeDefinition(String line) {
		return (line.contains("complexType") && line.contains("name="));
	}

	private boolean containstGroupReference(String line) {
		return (line.contains("group") && !line.contains("attributeGroup") && line.contains("ref="));
	}

	private boolean containsClosingGroupDefinition(String line) {
		return line.contains("</xsd:group>");
	}

	private boolean containsGroupDefinition(String line) {
		return (line.contains("group") && !line.contains("attributeGroup") && line.contains("name="));
	}

	private boolean containsClosingChoice(String line) {
		return line.contains("</xsd:choice>");
	}

	private boolean containsChoice(String line) {
		return line.contains("<xsd:choice");
	}

	private boolean containsClosingSequence(String line) {
		return line.contains("</xsd:sequence>");
	}

	private boolean containsSequence(String line) {
		return line.contains("<xsd:sequence");
	}

	private boolean containsOneLineElementDefinition(String line) {
		return (line.contains("element") && line.contains("name=") && line.contains("/>"));
	}

	private boolean containsOneLineElementReference(String line) {
		return (line.contains("element") && line.contains("ref=") && line.contains("/>"));
	}

	private boolean containsElementReferenceWithoutClosingSign(String line) {
		return (line.contains("element") && line.contains("ref=") && !line.contains("/>"));
	}

	private boolean containsClosingElementDefinition(String line) {
		return line.contains("</xsd:element>");
	}

	private boolean containsElementDefinition(String line) {
		return (line.contains("element") && line.contains("name=") && !line.contains("/>"));
	}

	private boolean containsDocumentation(String line) {
		return line.contains("documentation");
	}
}
